hezar/trainer/trainer.py

```diff
@@ -129,6 +129,25 @@ def _set_seed(seed) -> None:
         torch.manual_seed(seed)
         np.random.seed(seed)
         random.seed(seed)
+    
+    def info(self) -> None:
+        print("******************** Training Info ********************")
+        print(f"Task: {self.config.task}")
+        print(f"Model type: {type(self.model).__name__}")
+        print(f"Device(s): {self.device}")
+        print(f"Training Dataset: {self.train_dataset.config.name}")
+        print(f"Eval Dataset: {self.eval_dataset.config.name}")
+        print(f"Optimizer: {self.config.optimizer}")
```

Comment:
This better be 
```python
self.config.optimizer or self.default_optimizer
```
Since most of the time optimizer is not set by the user and the default one is used.